packages/fast-components-react-base/src/stack-panel/stack-panel.props.ts

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import {
+    ManagedClasses,
+    StackPanelClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+
+export interface StackPanelManagedClasses
+    extends ManagedClasses<StackPanelClassNameContract> {}
+export interface StackPanelUnhandledProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export enum stackPanelOrientation {
+    horizontal = "horizontal",
+    vertical = "vertical",
+}
+
+export interface ItemSpanOverride {
+    [key: number]: number;
+}
+
+export interface StackPanelHandledProps extends StackPanelManagedClasses {
+    /**
+     * Whether items outside the viewport should be rendered or not
+     * default is true
+     */
+    enableVirtualization?: boolean;
+
+    /**
+     *  the indexes of children that should always be rendered
+     */
+    neverVirtualizeIndexes?: number[];
+
+    /**
+     *  The default span of each item (width in horizontal orientation, height in vertical)
+     */
+    defaultItemSpan?: number;
+
+    /**
+     * An object whose entries specifies the height of items that differ from the default height.
+     * Format is {itemIndex: heightValue}
+     */
+    itemSpanOverrides?: ItemSpanOverride;
+
+    /**
+     * How many items to preload before and after the range defined by viewportMaxHeight
+     */
+    preloadBufferLength?: number;
+
+    /**
+     * The orientation of the stackpanel
+     */
+    orientation?: stackPanelOrientation;
+
+    /**
+     * The children of the panel
+     */
+    children?: React.ReactNode;
+
+    /**
+     * Callback for when scroll values change
+     */
+    onScrollChange?: (newScrollValue: number, scrollMaxValue: number) => void;
+
+    /**
+     * When the component scrolls focused content into view how many pixels should the next item "peek" into view
+     */
+    nextItemPeek?: number;
+
+    /**
+     * Whether to enable smooth scrolling
+     */
+    enableSmoothScrolling?: boolean;
```

Comment:
I wonder if we can't just provide a prop that is an animation function that can be passed, so instead of having a smooth scrolling function, we have a secondary smooth scrolling function they can pass in to a `scrollAnimation` prop

Reply:
I've thought about having a function to calculate the animation curve but I'm not sure we should do that instead of a Boolean.  I'm thinking we want the scrolling on by default but if all we have is a function prop then how do you turn it off?

Reply:
Well, this is in base correct? Animation is something I would consider as part of styling, so if anything I would offer a prop to inject the animation function, any implementation in a styled lib such as `fast-component-styles-msft` would then have the exported animation. Maybe @bheston or @nicholasrice has an opinion.

Reply:
As we chatted, I like the idea of not baking the animation into the component, though I understand the particular list / tabbing issue you're solving. If there's a good way to encapsulate this and allow it to be easily updated I think that makes sense.

Reply:
removed animation code as I found a css based solution